File: week 2/lcm/LCM.java
import java.util.*;

public class LCM {
  private static long gcd(long a, long b) {
	if (b == 0)
		return a;
	return gcd(b, a%b);
  }
	
  private static long lcm(long a, long b) {
    //write your code here
    return a * b / gcd(a,b);
  }

  public static void main(String args[]) {
    Scanner scanner = new Scanner(System.in);
    long a = scanner.nextLong();
    long b = scanner.nextLong();

    System.out.println(lcm(a, b));
  }
}

File: week 2/lcm/lcm.cpp
#include <iostream>

long long gcd(long long a, long long b) {
  //write your code here
  	if (b == 0)
		return a;
	return gcd(b, a%b);
}

long long lcm(long long a, long long b) {
  //write your code here
  return a*b / gcd(a,b);
}

int main() {
  long long a, b;
  std::cin >> a >> b;
  std::cout << lcm(a, b) << std::endl;
  return 0;
}
